nvtabular/io/dataset.py

```diff
@@ -861,7 +861,13 @@ def to_parquet(
                     fns = output_files[i : i + files_per_task]
                     start = i * split
                     stop = min(start + split * len(fns), ddf.npartitions)
-                    new[tuple(fns)] = np.arange(start, stop)
+                    if start < stop:
+                        new[tuple(fns)] = np.arange(start, stop)
+                if len(new.keys()) < output_files:
+                    warnings.warn(
+                        f"Only created {len(new.keys())} files did not have enough\n"
+                        f"partitions to create {len(output_files)} files."
+                    )
```

Comment:
Perhaps we can just change the heuristic we use to split partitions between files?

Somthing like this seems to work (although, I know its a bit hacky):
```python
            if isinstance(output_files, list):
                new = {}
                split = ddf.npartitions / len(output_files)
                splits = [math.floor(split), math.ceil(split)]
                ceil_freq = math.ceil(1/(split-splits[0])) if split-splits[1] else 1
                start = 0
                for i in range(0, len(output_files), files_per_task):
                    fns = output_files[i : i + files_per_task]
                    split = splits[1] if i % ceil_freq == 0 else splits[0]
                    stop = min(start + split * len(fns), ddf.npartitions)
                    new[tuple(fns)] = np.arange(start, stop)
                    start += split
                output_files = new
                suffix = ""  # Don't add a suffix later - Names already include it
```

Reply:
this fix fails on some of our unit tests. @rjzamora

Reply:
rerun tests